src/transformers/models/instructblip/modeling_instructblip.py

```diff
@@ -1271,10 +1266,13 @@ def __init__(self, config: InstructBlipConfig):
         self.qformer = InstructBlipQFormerModel(config.qformer_config)
 
         self.language_projection = nn.Linear(config.qformer_config.hidden_size, config.text_config.hidden_size)
+
         if config.use_decoder_only_language_model:
             language_model = AutoModelForCausalLM.from_config(config.text_config)
+            self._no_split_modules.append("LlamaDecoderLayer")
```

Comment:
No, should `language_model._no_split_modules` here.

Reply:
I think `language_model._no_split_modules` should already contain `LlamaDecoderLayer`, the purpose of adding it in `self._no_split_modules` is that in `from_pretrained` we never look at all the child modules that contain that attribute: https://github.com/huggingface/transformers/blob/195a9e5bdb1faa58cd58b47a23a47734d2b90d8c/src/transformers/modeling_utils.py#L2807C13-L2807C29 | then gets passed to accelerate this way: https://github.com/huggingface/transformers/blob/195a9e5bdb1faa58cd58b47a23a47734d2b90d8c/src/transformers/modeling_utils.py#L2814 right? So not sure we should add it to `language_model`. What do you think?

Reply:
I don't understand your comment. The code loads any model from the Hub (granted it should be llama models in the actual checkpoints) but you add the specific Llama no split blocks. This will stop working if someone adds an instructblip-2 model that loads a different language model.

Reply:
A cleaner implementation would be that in `post_init` we should add an utility method to look at all child attributes that contain `_no_split_modules` and dynamically append them to `self._no_split_modules`

Reply:
If you think that's better, happy to change the PR to add these changes

Reply:
Ah I see yes, sorry I misunderstood your comment, will update that now